### DISCUSSION TITLE
How to extract all text objects from .pdf to .txt?

### DISCUSSION THREAD

**Discussion by Korb** [qpdf/qpdf#892] [Q&A]
Interested in the ability to view in plain text format all text objects displayed on the pages of a PDF document or present on an invisible text layer (searchable image PDF).

**Comment by jberkenbilt** [comment]
This is not something qpdf can do. I recommend something either from poppler or ghostscript. PDF text extraction requires a full PDF interpreter that knows how to do page rendering, which is quite far out of scope for qpdf. What qpdf can do is expose the raw PDF code to render the page. In some simple documents, the text is readable, but in most machine-generated PDF files, the PDF text is not in human-readable form.
